pkg/distro/generic/images.go

```diff
@@ -47,13 +48,9 @@ func osCustomizations(t *imageType, osPackageSet rpmmd.PackageSet, options distr
 		// an argument as fallback or make it not return the standard "kernel"
 		// when it's unset.
 		osc.KernelName = c.GetKernel().Name
-		if imageConfig.DefaultKernelName != nil {
-			osc.KernelName = *imageConfig.DefaultKernelName
-		}
+		osc.KernelName = common.ValueOrEmpty(imageConfig.DefaultKernelName)
```

Comment:
We can't just simply use the `common.ValueOrEmpty()` everywhere, because it overrides the potentially previously set value with the type's default value. In this case, the kernel package name gets overridden with an empty string.